### PR TITLE
SUS-2395: Cleanup service classes

### PR DESCRIPTION
**Pull request by mszabo-wikia**
* remove useless `Service` class
* remove `extends WikiaService` clause for non-Nirvana classes
* remove unused services

https://wikia-inc.atlassian.net/browse/SUS-2395

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 25 issues
* ![MAJOR][MAJOR] 25 major

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [LyricFindTrackingService.class.php#L148](https://github.com/Wikia/app/blob/6913368b0561419af0871ad03e06fb01836879fc/extensions/3rdparty/LyricWiki/LyricFind/LyricFindTrackingService.class.php#L148): Reduce this "switch/case" number of lines from 11 to at most 10, for example by extracting code into function. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1151)
1. ![MAJOR][MAJOR] [ArticleAsJson.class.php#L286](https://github.com/Wikia/app/blob/6913368b0561419af0871ad03e06fb01836879fc/extensions/wikia/ArticleAsJson/ArticleAsJson.class.php#L286): Remove the unused function parameter "$dummy". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [ArticleAsJson.class.php#L288](https://github.com/Wikia/app/blob/6913368b0561419af0871ad03e06fb01836879fc/extensions/wikia/ArticleAsJson/ArticleAsJson.class.php#L288): Remove the unused function parameter "$file". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [ArticleAsJson.class.php#L291](https://github.com/Wikia/app/blob/6913368b0561419af0871ad03e06fb01836879fc/extensions/wikia/ArticleAsJson/ArticleAsJson.class.php#L291): Remove the unused function parameter "$time". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [ArticleAsJson.class.php#L414](https://github.com/Wikia/app/blob/6913368b0561419af0871ad03e06fb01836879fc/extensions/wikia/ArticleAsJson/ArticleAsJson.class.php#L414): Remove the unused function parameter "$parser". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [ArticleAsJson.class.php#L433](https://github.com/Wikia/app/blob/6913368b0561419af0871ad03e06fb01836879fc/extensions/wikia/ArticleAsJson/ArticleAsJson.class.php#L433): Remove the unused function parameter "$parser". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [CategoryService.class.php#L45](https://github.com/Wikia/app/blob/6913368b0561419af0871ad03e06fb01836879fc/extensions/wikia/CategoryGalleries/services/CategoryService.class.php#L45): Remove the unused function parameter "$tokens". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [CategoryService.class.php#L270](https://github.com/Wikia/app/blob/6913368b0561419af0871ad03e06fb01836879fc/extensions/wikia/CategoryGalleries/services/CategoryService.class.php#L270): Remove the unused function parameter "$newtitle". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [CategoryService.class.php#L270](https://github.com/Wikia/app/blob/6913368b0561419af0871ad03e06fb01836879fc/extensions/wikia/CategoryGalleries/services/CategoryService.class.php#L270): Remove the unused function parameter "$user". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [EditPageService.class.php#L23](https://github.com/Wikia/app/blob/6913368b0561419af0871ad03e06fb01836879fc/extensions/wikia/EditPageLayout/EditPageService.class.php#L23): Remove the unused function parameter "$request". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 25 issues
* ![MAJOR][MAJOR] 25 major

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [LyricFindTrackingService.class.php#L148](https://github.com/Wikia/app/blob/92db2e60b12d3920e6d988a440bc417b34c0097c/extensions/3rdparty/LyricWiki/LyricFind/LyricFindTrackingService.class.php#L148): Reduce this "switch/case" number of lines from 11 to at most 10, for example by extracting code into function. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1151)
1. ![MAJOR][MAJOR] [ArticleAsJson.class.php#L286](https://github.com/Wikia/app/blob/92db2e60b12d3920e6d988a440bc417b34c0097c/extensions/wikia/ArticleAsJson/ArticleAsJson.class.php#L286): Remove the unused function parameter "$dummy". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [ArticleAsJson.class.php#L288](https://github.com/Wikia/app/blob/92db2e60b12d3920e6d988a440bc417b34c0097c/extensions/wikia/ArticleAsJson/ArticleAsJson.class.php#L288): Remove the unused function parameter "$file". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [ArticleAsJson.class.php#L291](https://github.com/Wikia/app/blob/92db2e60b12d3920e6d988a440bc417b34c0097c/extensions/wikia/ArticleAsJson/ArticleAsJson.class.php#L291): Remove the unused function parameter "$time". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [ArticleAsJson.class.php#L414](https://github.com/Wikia/app/blob/92db2e60b12d3920e6d988a440bc417b34c0097c/extensions/wikia/ArticleAsJson/ArticleAsJson.class.php#L414): Remove the unused function parameter "$parser". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [ArticleAsJson.class.php#L433](https://github.com/Wikia/app/blob/92db2e60b12d3920e6d988a440bc417b34c0097c/extensions/wikia/ArticleAsJson/ArticleAsJson.class.php#L433): Remove the unused function parameter "$parser". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [CategoryService.class.php#L45](https://github.com/Wikia/app/blob/92db2e60b12d3920e6d988a440bc417b34c0097c/extensions/wikia/CategoryGalleries/services/CategoryService.class.php#L45): Remove the unused function parameter "$tokens". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [CategoryService.class.php#L270](https://github.com/Wikia/app/blob/92db2e60b12d3920e6d988a440bc417b34c0097c/extensions/wikia/CategoryGalleries/services/CategoryService.class.php#L270): Remove the unused function parameter "$newtitle". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [CategoryService.class.php#L270](https://github.com/Wikia/app/blob/92db2e60b12d3920e6d988a440bc417b34c0097c/extensions/wikia/CategoryGalleries/services/CategoryService.class.php#L270): Remove the unused function parameter "$user". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [EditPageService.class.php#L23](https://github.com/Wikia/app/blob/92db2e60b12d3920e6d988a440bc417b34c0097c/extensions/wikia/EditPageLayout/EditPageService.class.php#L23): Remove the unused function parameter "$request". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: includes/wikia/services/InspectletService.class.php [hunk 1]
```diff
@@ -1,32 +0,0 @@
-<?php
-
-/**
- * Class InspectletService
- * This service provides interface to add Inspectlet integrations to parts of the product
- *
- * NOTE! This service is disabled! If you want to use it, load it in `DefaultSettings.php` file.
```
Comment (macbre) [inline includes/wikia/services/InspectletService.class.php:7]: What ?!

File: extensions/wikia/JsonFormat/JsonFormatService.php [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
 	private $htmlParser;
 	private $requestContext;
 
-	function __construct( $htmlParser = null ) {
-		if( $htmlParser == null ) {
-			$htmlParser = new Wikia\JsonFormat\HtmlParser();
-		}
-		$this->requestContext = new RequestContext();
-		$this->htmlParser = $htmlParser;
+	public function __construct( HtmlParser $htmlParser = null ) {
```
Comment (macbre) [inline extensions/wikia/JsonFormat/JsonFormatService.php]: The only use of this class does not pass the `$htmlParser` argument:

```php
includes/wikia/api/ArticlesApiController.class.php
981:		$jsonFormatService = new JsonFormatService();
```

Maybe let's just remove it?

And there's:

```
extensions/wikia/Search/classes/IndexService/DefaultContent.php
9:use JsonFormatService;
```

but no use of this class within the file.

**Review by macbre — CHANGES_REQUESTED** [review]
One inline comment left

**Comment by WikiaTech** [comment]
SonarQube analysis reported 25 issues
* ![MAJOR][MAJOR] 25 major

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [LyricFindTrackingService.class.php#L148](https://github.com/Wikia/app/blob/510253bbf6c1f59efe816a1ee2d9640bc9a6b63b/extensions/3rdparty/LyricWiki/LyricFind/LyricFindTrackingService.class.php#L148): Reduce this "switch/case" number of lines from 11 to at most 10, for example by extracting code into function. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1151)
1. ![MAJOR][MAJOR] [ArticleAsJson.class.php#L286](https://github.com/Wikia/app/blob/510253bbf6c1f59efe816a1ee2d9640bc9a6b63b/extensions/wikia/ArticleAsJson/ArticleAsJson.class.php#L286): Remove the unused function parameter "$dummy". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [ArticleAsJson.class.php#L288](https://github.com/Wikia/app/blob/510253bbf6c1f59efe816a1ee2d9640bc9a6b63b/extensions/wikia/ArticleAsJson/ArticleAsJson.class.php#L288): Remove the unused function parameter "$file". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [ArticleAsJson.class.php#L291](https://github.com/Wikia/app/blob/510253bbf6c1f59efe816a1ee2d9640bc9a6b63b/extensions/wikia/ArticleAsJson/ArticleAsJson.class.php#L291): Remove the unused function parameter "$time". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [ArticleAsJson.class.php#L414](https://github.com/Wikia/app/blob/510253bbf6c1f59efe816a1ee2d9640bc9a6b63b/extensions/wikia/ArticleAsJson/ArticleAsJson.class.php#L414): Remove the unused function parameter "$parser". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [ArticleAsJson.class.php#L433](https://github.com/Wikia/app/blob/510253bbf6c1f59efe816a1ee2d9640bc9a6b63b/extensions/wikia/ArticleAsJson/ArticleAsJson.class.php#L433): Remove the unused function parameter "$parser". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [CategoryService.class.php#L45](https://github.com/Wikia/app/blob/510253bbf6c1f59efe816a1ee2d9640bc9a6b63b/extensions/wikia/CategoryGalleries/services/CategoryService.class.php#L45): Remove the unused function parameter "$tokens". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [CategoryService.class.php#L270](https://github.com/Wikia/app/blob/510253bbf6c1f59efe816a1ee2d9640bc9a6b63b/extensions/wikia/CategoryGalleries/services/CategoryService.class.php#L270): Remove the unused function parameter "$newtitle". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [CategoryService.class.php#L270](https://github.com/Wikia/app/blob/510253bbf6c1f59efe816a1ee2d9640bc9a6b63b/extensions/wikia/CategoryGalleries/services/CategoryService.class.php#L270): Remove the unused function parameter "$user". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [EditPageService.class.php#L23](https://github.com/Wikia/app/blob/510253bbf6c1f59efe816a1ee2d9640bc9a6b63b/extensions/wikia/EditPageLayout/EditPageService.class.php#L23): Remove the unused function parameter "$request". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by macbre — APPROVED** [review]

**Pull request merged**